openmc/lattice.py

```diff
@@ -1094,16 +1187,40 @@ def get_local_coordinates(self, point, idx):
             system
 
         """
-        x = point[0] - (self.center[0] + sqrt(0.75)*self.pitch[0]*idx[0])
-        y = point[1] - (self.center[1] + (0.5*idx[0] + idx[1])*self.pitch[0])
+        if self._hextype:#//DR
+           x = point[0] - (self.center[0] + self.pitch[0]*idx[0] + 0.5*self.pitch[0]*idx[1])
+           y = point[1] - (self.center[1] + (sqrt(0.75)*self.pitch[0]*idx[1]))
+        else:
+           x = point[0] - (self.center[0] + sqrt(0.75)*self.pitch[0]*idx[0])
+           y = point[1] - (self.center[1] + (0.5*idx[0] + idx[1])*self.pitch[0])
+
         if self._num_axial is None:
             z = point[2]
         else:
             z = point[2] - (self.center[2] + (idx[2] + 0.5 - 0.5*self.num_axial)*
                             self.pitch[1])
         return (x, y, z)
+    def get_universe_index(self, idx): #//DR
+        r"""Return index in the universes array corresponding to a lattice element index
 
-    def get_universe_index(self, idx):
+        Parameters
+        ----------
+        idx : Iterable of int
+            Lattice element indices in the :math:`(x,\alpha,z)` coordinate
+            system in 'OY' orientation case, or indices in the :math:`(\alpha,y,z)` coordinate
+            system in 'OX' one 
+
+        Returns
+        -------
+        2- or 3-tuple of int
+            Indices used when setting the :attr:`HexLattice.universes` property
+
+        """
+        if self._hextype:
+           return self.get_universe_index_ox(idx)
+        else:
+           return self.get_universe_index_oy(idx)
+    def get_universe_index_oy(self, idx):#//DR
```

Comment:
Put a blank line before this function.

This function should never be called by the user (they'll use get_universe_index instead) so make it and the ox version private by prepending and underscore to the name like `def _get_universe_index_oy`